### PR TITLE
Fix: Classes are not applied to elements after rename [ED-20634]

### PR DESCRIPTION
**Pull request by Omerisra6**
### ✨ PR Description
Purpose: Fix class application to elements by restructuring the class transformation logic to properly handle renamed CSS classes.

Main changes:
- Added caching mechanism to store provider keys for class IDs to improve performance
- Restructured transformClassId function to retrieve and apply resolved CSS names
- Added comprehensive test suite for classes transformer with various edge cases

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using. **[We'd love your feedback!](mailto:product@linearb.io)** 🚀</sub>

### CODE REVIEW

**Review by MiryamOren — DISMISSED** [review]

File: packages/packages/core/editor-canvas/src/transformers/settings/classes-transformer.ts [hunk 1]
```diff
@@ -2,30 +2,30 @@ import { stylesRepository } from '@elementor/editor-styles-repository';
...
+		}
+
+		cache.set( id, provider.getKey() );
+	}
+
+	const providerKey = cache.get( id ) as string;
+
+	const provider = stylesRepository.getProviderByKey( providerKey );
+
+	return provider?.actions.resolveCssName( id ) || id;
```
Comment (MiryamOren) [inline packages/packages/core/editor-canvas/src/transformers/settings/classes-transformer.ts]:
```suggestion
	return provider?.actions.resolveCssName( id ) ?? id;
```

**Review by MiryamOren — DISMISSED** [review]

**Review by MiryamOren — DISMISSED** [review]

**Review by MiryamOren — APPROVED** [review]
